tests/test_operator.py

```diff
@@ -950,6 +953,37 @@ def test_incomplete_override(self):
 
 class TestDeclarator(object):
 
+    def test_conditional_declarations(self):
+        accesses = [[0, 0], [0, 1], [1, 0], [1, 1]]
+        dims = len(accesses[0])
+        pts = len(accesses)
+        stencil_name = 's%dd_%s_%dpt' % (dims, 'name', pts)
+        stencil_array = Array(
+            name=stencil_name,
+            dimensions=(DefaultDimension(name='len', default_value=dims * pts),),
```

Comment:
do we really need DefaultDimension ?

Reply:
It is not possible to pass a tuple directly:

```
return super().free_symbols - {d for d in self.dimensions if d.is_Default}
E   AttributeError: 'tuple' object has no attribute 'is_Default' 
```